plugins/arrow-flight-rpc/build.gradle

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+apply plugin: 'opensearch.internal-cluster-test'
+
+opensearchplugin {
+  description = 'Arrow flight based Stream implementation'
+  classname = 'org.opensearch.arrow.flight.FlightStreamPlugin'
+}
+
+dependencies {
+  implementation project(':libs:opensearch-arrow-spi')
+  implementation project(path:':libs:opensearch-flight-core-shaded', configuration: 'shadow')
```

Comment:
The plugin could bundle all these dependencies directly, right? There is no need to shade as far as we don't have netty / jackson JarHell (that we shouldn't)

Reply:
plugin has a different classloader than server, so even if plugin adds them to their classloader, it won't work for server at runtime.